### PR TITLE
Determine maximum number of bits for record ID based on current store format

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Remove hardcoded number of bits for each particular id type;
use current store format to determine allowable max id for each particular record type

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@tinwelint rebased

File: community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java [hunk 1]
```diff
@@ -820,7 +795,7 @@ private boolean arrayContains( long[] ids, int cursor, int labelId )
     @Override
     public void createNode( long id, Map<String, Object> properties, Label... labels )
     {
-        if ( id < 0 || id > MAX_NODE_ID )
+        if ( id < 0 || id > recordFormats.node().getMaxId() )
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java]: The RecordFormats methods weren't designed to be called in hot method like this one. For convenience those methods simply return a new RecordFormat instance every time. Rather figure out the max ID when constructing the BatchInserter and use that final value here instead.

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java]: absolutely true!

**Comment by MishaDemianenko** [comment]
@tinwelint comments addressed

**Pull request merged**